bin/annotation_workflow.py

```diff
@@ -0,0 +1,76 @@
+import json, re
+from pathlib import Path
+
+def github_action_notice(file, url, line):
+    """
+    Print GitHub action notice with file path, URL, and line number.
+
+    Parameters:
+        file (str): File path.
+        url (str): URL.
+        line (int): Line number.
+    """
+    return f"::warning file={file},line={line}::Annotation: {url}"
+
+def find_test_line_number(test_content, test_name):
+    """
+    Find the line number of a test in the JSON content.
+
+    Parameters:
+        test_content (str): JSON content.
+        test_name (str): Test name.
+
+    Returns:
+        int: Line number of the test.
+    """
+    lines = test_content.split("\n")  # Split content into lines
+    for i, line in enumerate(lines, start=1):  # Iterate over lines
+        if test_name in line:  # Check if test name is found in the line
+            return i  # Return the line number if found
+    return 1  # Return None if test name is not found
+
+def clear_previous_annotations():
+    """
+    Clear previous GitHub action annotations.
+    """
+    print("::remove-matcher owner=me::")
+
+# Specify the path to the JSON file using pathlib.Path
```

Comment:
I'd remove these comments personally, they don't seem to add anything more than what the code says.